lib/plausible/installation_support/checks/installation.ex

```diff
@@ -22,9 +45,24 @@ defmodule Plausible.InstallationSupport.Checks.Installation do
       return await page.evaluate(async (expectedDataDomain, debug) => {
         return await window.verifyPlausibleInstallation(expectedDataDomain, debug);
       }, context.expectedDataDomain, context.debug);
-    } catch (error) {
-      const msg = error.message ? error.message : JSON.stringify(error)
-      return {data: {completed: false, error: msg}}
+    }
+
+    for (let attempt = 0; attempt <= MAX_RETRIES; attempt++) {
+      try {
+        const isFirstAttempt = attempt === 0
+        return await attemptVerification(isFirstAttempt)
+      } catch (error) {
+        const shouldRetry = typeof error.message === 'string' && error.message.toLowerCase().includes('execution context')
```

Comment:
suggestion, non-blocking: since error might be undefined in JS, we should access its fields defensively
```suggestion
        const shouldRetry = typeof error?.message === 'string' && error.message.toLowerCase().includes('execution context')
```